pageserver/src/tenant/timeline/walreceiver/connection_manager.rs

```diff
@@ -869,6 +871,14 @@ fn wal_stream_connection_config(
     Ok(connstr)
 }
 
+// Takes availability_zone from proto structure and converts empty string to None.
+fn availability_zone_from_broker_info(info: &SafekeeperTimelineInfo) -> Option<String> {
```

Comment:
Sorry for being tedious, but probably just add `optional` to the proto? It is a bit strange to convert, I'd rather have everywhere option or have everywhere empty string...

Reply:
Apparently, I have an old protoc on my fresh ubuntu installation. That means that others can have issues with it as well:
```
error: failed to run custom build command for `storage_broker v0.1.0 (/workspaces/codespaces-blank/neon/storage_broker)`

Caused by:
  process didn't exit successfully: `/workspaces/codespaces-blank/neon/target/debug/build/storage_broker-204108b4c82a3da8/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=proto/broker.proto
  cargo:rerun-if-changed=proto

  --- stderr
  thread 'main' panicked at 'failed to compile protos Custom { kind: Other, error: "protoc failed: broker.proto:40:14: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.\n" }', storage_broker/build.rs:9:29
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
make: *** [Makefile:66: neon] Error 101
@petuhovskiy ➜ /workspaces/codespaces-blank/neon (walreceiver-same-az) $ protoc --version
libprotoc 3.6.1
```

I actually don't think optional is a good idea, because I'm cautious that some grpc clients can still send empty string, even if there will be `optional` in the proto. But I'll try to upgrade my protoc to see what it will look like.

Reply:
I'm on debian 12, and here it is fresh enough (3.21). But I see that there is issue with older distributions, so yes, seems like manual install will be required. Anyway, I'm fine with either variant.